libkbfs/folder_branch_ops.go

```diff
@@ -2543,16 +2542,20 @@ func (fbo *folderBranchOps) createEntryLocked(
 			Ctime: now,
 		},
 	}
-	fbo.blocks.AddDirEntryInCache(lState, dirPath, name, de)
+
+	dirCacheUndoFn := fbo.blocks.AddDirEntryInCache(lState, dirPath, name, de)
 	fbo.dirOps = append(fbo.dirOps, cachedDirOp{co, []Node{dir, node}})
+	added := fbo.status.addDirtyNode(dir)
 
 	cleanupFn := func() {
-		// Until KBFS-2076 is done, clear out the cached dir data manually.
-		fbo.dirOps = nil
-		fbo.blocks.ClearCachedAddsAndRemoves(lState, dirPath)
-		fbo.blocks.ClearCacheInfo(lState, dirPath.ChildPath(name, newPtr))
+		if added {
+			fbo.status.rmDirtyNode(dir)
+		}
+		fbo.dirOps = fbo.dirOps[:len(fbo.dirOps)-1]
+		dirCacheUndoFn(lState)
 	}
-	defer cleanupFn()
+	// Until KBFS-2076 is done, always clear out the cached dir data.
+	defer func() { cleanupFn() }()
```

Comment:
Why is this nested in another `func()`?

Hmm I see now, you assign it to `func() {}` below. Can you use `nil` for this, with a `nil` check in the function? Calling empty closures takes a lot longer than a `nil` check.